core/src/letter_distribution.c

```diff
@@ -114,23 +118,66 @@ void load_letter_distribution(LetterDistribution *letter_distribution,
 
     machine_letter++;
   }
-
+  letter_distribution->max_tile_length = max_tile_length;
   fclose(file);
 }
 
-// Assumes english
+// This is a linear search. This function should not be used for anything
+// that is speed-critical. If we ever need to use this in anything
+// speed-critical, we should use a hash.
 uint8_t
 human_readable_letter_to_machine_letter(LetterDistribution *letter_distribution,
-                                        unsigned char r) {
-  return (uint8_t)
-      letter_distribution->human_readable_letter_to_machine_letter[r];
+                                        char *letter) {
+
+  for (int i = 0; i < MACHINE_LETTER_MAX_VALUE; i++) {
+    if (strcmp(letter_distribution->machine_letter_to_human_readable_letter[i],
+               letter) == 0) {
+      return i;
+    }
+  }
+  // It better find it
```

Comment:
Should we be aborting/asserting or otherwise fatally exiting here if we always expect to find the letter?

Reply:
I'll delete this comment; it's actually ok to return INVALID_LETTER (see algo below)